project/src/main/java/com/google/sps/data/DataAccessor.java

```diff
@@ -70,7 +72,26 @@ public boolean isRegistered(String userId) {
   public void registerUser(String userId) {
     checkArgument(!Strings.isNullOrEmpty(userId), "Invalid user ID");
     checkArgument(!isRegistered(userId), "User already registered.");
-    Entity userEntity = new Entity(USER_ENTITY_NAME, userId);
+    Entity userEntity = new Entity(USER_ENTITY_KIND, userId);
     datastoreService.put(userEntity);
   }
+
+  /**
+   * Stores the UserPreferences in the personalized user database.
+   *
+   * @param userId The ID of the user to store the preferred cuisines for.
+   * @param userPref The user choices on the query form to store in the user’s database.
+   */
+  public void storeUserPreferences(String userId, UserPreferences userPref) {
+    checkArgument(!Strings.isNullOrEmpty(userId), "Invalid user ID");
+    if (userPref.cuisines().isEmpty()) {
```

Comment:
I'd do:
if(!userPref.cuisines().isEmpty()){// logic)}

Reply:
I like it better too :)